scripts/quests/otherAreas/The_Gift.lua

```diff
@@ -0,0 +1,93 @@
+-----------------------------------
+-- The Gift
+-----------------------------------
+-- Log ID: 4, Quest ID: 21
+-- Oswald  : !pos 47.119 -15.273 7.989 248
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.THE_GIFT)
+
+quest.reward =
+{
+    item = xi.items.SLEEP_DAGGER,
+    title = xi.title.SAVIOR_OF_LOVE
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.UNDER_THE_SEA) == QUEST_COMPLETED and
+            player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.THE_SAND_CHARM) >= QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SELBINA] =
+        {
+            ['Oswald'] =
+            {
+                onTrigger = quest:event(70, xi.items.DANCESHROOM)
+            },
+
+            onEventFinish =
+            {
+                [70] = function(player, csid, option, npc)
+                    if option == 50 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SELBINA] =
+        {
+            ['Oswald'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHas(trade, xi.items.DANCESHROOM) then
```

Comment:
should make trades npcUtil.tradeHasExactly, else players can trade other items along with it and lose those items! :O